commit 756b2abf0d4db86052b09576613b2fa0ce4d435b
Author: BinhNV <binhnv@exoplatform.com>
Date:   Fri Jul 13 16:46:24 2012 +0700

    modified file

diff --git a/src/main/java/exoplatform/BookStoreService.java b/src/main/java/exoplatform/BookStoreService.java
index 5c265d5..8e45abd 100644
--- a/src/main/java/exoplatform/BookStoreService.java
+++ b/src/main/java/exoplatform/BookStoreService.java
@@ -17,7 +17,6 @@
 package exoplatform;
 
 import java.io.Serializable;
-import java.util.ArrayList;
 import java.util.List;
 
 import javax.jcr.RepositoryException;
@@ -33,7 +32,6 @@ import exoplatform.entity.Author;
 import exoplatform.entity.Book;
 import exoplatform.entity.Book.CATEGORY;
 import exoplatform.exception.DuplicateBookException;
-import exoplatform.utils.Utils;
 
 /**
  * Created by The eXo Platform SAS
@@ -149,8 +147,8 @@ public class BookStoreService implements Startable {
       addBook("Hibernate in action", CATEGORY.TECHNICAL, "Hibernate");
       addBook("Napoleon", CATEGORY.HISTORY, "Napoleon");
       addBook("Jouney to the West", CATEGORY.COMICS, "Jouney to the West");
-    } catch (DuplicateBookException e) {
-      e.printStackTrace();
+    } catch (DuplicateBookException de) {
+      log.error("Duplicate book", de);
     }
   } 
   
diff --git a/target/classes/exoplatform/BookStoreService.class b/target/classes/exoplatform/BookStoreService.class
index 04e4b93..a0ea9a7 100644
Binary files a/target/classes/exoplatform/BookStoreService.class and b/target/classes/exoplatform/BookStoreService.class differ
